### ISSUE TITLE
Create device with PSK instead of serialnumbers

### ISSUE THREAD

**Issue by esecesec** [tmorris-ftnt/ztptool#22]
Can´t see it´s possible to create a device without serialnumber and instead use a key. Would that be possible somehow?

**Comment by tmorris-ftnt** [comment]
Correct, it’s not supported in the current build. 

I’ll have a look and see if it can be implemented easily.

**Comment by esecesec** [comment]
Thanks! We are going to roll out around 3000 Fortigates, so this would really help us out if it was possible :)

**Comment by tmorris-ftnt** [comment]
sounds like an interesting project.

i've made a quick test build here

https://github.com/tmorris-ftnt/tmorris-ftnt.github.io/raw/master/ztptool-v1.0.15psk-beta-win.zip

to use this in the Device_SN column put "PSK" then create a new column called "Device_PSK" and put the PSK you wish to use in there..

**Comment by esecesec** [comment]
That worked really good, huge thanks :)

However, it doesn´t seem to be able to assign a pre-run CLI template (FMG 7.0.4) and due to that the policy package fails to install. 

"Assign CLI Template "SV_PreRun_40F" successful"
"Install policy package "PP-SV" failed"

If we assign the prerun and policy package after the import, everything works but we then need to run the install wizard for the policy package. 

Thanks!

**Comment by tmorris-ftnt** [comment]
does the task monitor with in FMG give you any clue as to why the policy package install fails?

Is there an SD-WAN or VPN template or anything involved?

**Comment by esecesec** [comment]
Yes, it is depending on interfaces we create in pre-run scripts so it fails due to that not assigned. We also have a template group contain both CLI, System and SD-WAN.

**Comment by tmorris-ftnt** [comment]
Ok, im not really sure with out looking at it. You can drop me a line via email (tmorris at fortinet.com) and send me more info (the xlsx file for example) or we can setup a screen share if you like.
